File: python/01-circles-demo/py/step-03.py
# Demo 1: Add Randomness to Size and Color (Processing.py)
# Step 3: Randomized grid of circles
# For use in Processing IDE, Python mode.

def setup():
    size(600, 600)
    background(255)
    noStroke()
    cols = 10
    rows = 10
    diameter = 40
    padding = 20
    for y in range(rows):
        for x in range(cols):
            cx = padding + x * (diameter + padding)
            cy = padding + y * (diameter + padding)
            # Randomize diameter and color
            d = random(20, 60)
            r = random(60, 200)
            g = random(120, 220)
            b = random(200, 255)
            fill(r, g, b)
            ellipse(cx, cy, d, d)

File: javascript/01-circles-demo/js/step-03.js

// Step 3: Draw a grid of circles using nested loops
// This sketch shows how to use two for loops to draw a grid

function setup() {
  createCanvas(800, 800); // Create a canvas 800x800 pixels
  background(220); // Light gray background
  noStroke(); // No outline on circles
  fill(100, 150, 255); // Set fill color
  let numCols = 8; // Number of columns
  let numRows = 8; // Number of rows
  let spacingX = width / (numCols + 1); // Horizontal spacing
  let spacingY = height / (numRows + 1); // Vertical spacing
  for (let i = 1; i <= numCols; i++) {
    for (let j = 1; j <= numRows; j++) {
      ellipse(i * spacingX, j * spacingY, 60); // Draw each circle in the grid
    }
  }
}

// No animation in this step
